std/digest/crc.d

```diff
@@ -138,11 +135,53 @@ private static immutable uint[256][8] crc32Tables = genTables(0xEDB88320);
  * Template API CRC32 implementation.
  * See $(D std.digest.digest) for differences between template and OOP API.
  */
-struct CRC32
+alias CRC32 = CRC!(32, 0xEDB88320);
+
+/**
+ * Template API CRC64-ECMA implementation.
+ * See $(D std.digest.digest) for differences between template and OOP API.
+ */
+alias CRC64ECMA = CRC!(64, 0xC96C5795D7870F42);
+
+/**
+ * Template API CRC64-ISO implementation.
+ * See $(D std.digest.digest) for differences between template and OOP API.
+ */
+alias CRC64ISO = CRC!(64, 0xD800000000000000);
+
+/**
+ * Generic Template API used for CRC32 and CRC64 implementations.
+ *
+ * The N parameter indicate the size of the hash in bits.
+ * The parameter P specify the polynomial to be used for reduction.
+ *
+ * You may want to use the CRC32, CRC65ECMA and CRC64ISO aliases
+ * for convenience.
+ *
+ * See $(D std.digest.digest) for differences between template and OOP API.
+ */
+struct CRC(uint N, ulong P) if (N == 32 || N == 64)
 {
     private:
+        static if (N == 32)
+        {
+            alias T = uint;
+        }
+        else
+        {
+            alias T = ulong;
+        }
+
+        static immutable T[256][8] tables = genTables!T(P);
+
+        /**
+         * Type of the finished CRC hash.
+         * ubyte[8] if N is 32, ubyte[16] if N is 64.
+         */
+        alias R = ubyte[T.sizeof];
```

Comment:
Shouldn't `R` be `ubyte[4]` for 32 bit? Because `finish` used to return `ubyte[4]` originally and now it returns `ubyte[8]`.